### PR TITLE
refactor cmd-storage test

### PR DESCRIPTION
**Pull request by anandrkskd**
Signed-off-by: anandrkskd <anandrkskd@gmail.com>

**What type of PR is this?**
/kind code-refactoring

**What does this PR do / why we need it**:

Refactor tests in tests/integration/cmd_storage_test.go  file

### CODE REVIEW

File: tests/integration/cmd_storage_test.go [hunk 1]
```diff
@@ -23,56 +23,95 @@ var _ = Describe("odo storage command tests", func() {
...
-			Expect(StorageList).To(ContainSubstring("Pushed"))
-
-			// Delete storage, list storage should have state "Locally Deleted"
-			helper.Cmd("odo", "storage", "delete", "pv1", "-f", "--context", commonVar.Context).ShouldPass()
-			StorageList = helper.Cmd("odo", "storage", "list", "--context", commonVar.Context).ShouldPass().Out()
-			Expect(StorageList).To(ContainSubstring("Locally Deleted"))
-
+		When("creating storage", func() {
+			BeforeEach(func() {
+				// create storage, list storage should have state "Not Pushed"
```
Comment (feloy) [inline tests/integration/cmd_storage_test.go:86]: you can remove this comment, this following It is explicit enough

File: tests/integration/cmd_storage_test.go [hunk 2]
```diff
@@ -23,56 +23,95 @@ var _ = Describe("odo storage command tests", func() {
...
+				helper.Cmd("odo", "storage", "create", "pv1", "--path=/tmp1", "--size=1Gi", "--context", commonVar.Context).ShouldPass()
+			})
+
+			It("should list storage as Not Pushed", func() {
+
+				StorageList := helper.Cmd("odo", "storage", "list", "--context", commonVar.Context).ShouldPass().Out()
+				Expect(StorageList).To(ContainSubstring("Not Pushed"))
+			})
+			When("storage is pushed", func() {
+				// Push storage, list storage should have state "Pushed"
```
Comment (feloy) [inline tests/integration/cmd_storage_test.go:96]: Idem

File: tests/integration/cmd_storage_test.go [hunk 3]
```diff
@@ -23,56 +23,95 @@ var _ = Describe("odo storage command tests", func() {
...
+					StorageList := helper.Cmd("odo", "storage", "list", "--context", commonVar.Context).ShouldPass().Out()
+					Expect(StorageList).To(ContainSubstring("Pushed"))
+
+				})
+				When("storage is deleted", func() {
+					BeforeEach(func() {
+						helper.Cmd("odo", "storage", "delete", "pv1", "-f", "--context", commonVar.Context).ShouldPass()
+					})
+					It("should have state Locally Deleted", func() {
+						// Delete storage, list storage should have state "Locally Deleted"
```
Comment (feloy) [inline tests/integration/cmd_storage_test.go:109]: Idem

File: tests/integration/cmd_storage_test.go [hunk 4]
```diff
@@ -23,56 +23,95 @@ var _ = Describe("odo storage command tests", func() {
...
+			})
+
+			It("should list storage as Not Pushed", func() {
+
+				StorageList := helper.Cmd("odo", "storage", "list", "--context", commonVar.Context).ShouldPass().Out()
+				Expect(StorageList).To(ContainSubstring("Not Pushed"))
+			})
+			When("storage is pushed", func() {
+				// Push storage, list storage should have state "Pushed"
+				helper.Cmd("odo", "push", "--context", commonVar.Context).ShouldPass()
```
Comment (feloy) [inline tests/integration/cmd_storage_test.go:97]: This should be in a BeforeEach

File: tests/integration/cmd_storage_test.go (hunk 1 above)
Comment (anandrkskd) [reply tests/integration/cmd_storage_test.go]: understood, I will remove the comments!

Comment (anandrkskd) [reply tests/integration/cmd_storage_test.go]: Updated!

**Comment by feloy** [comment]
/test v4.7-integration-e2e

flaky test
[Fail] odo devfile status command tests Verify devfile status works [It] Verify that odo component status correctly reports supervisord status

File: tests/integration/cmd_storage_test.go [hunk 5]
```diff
@@ -23,56 +23,93 @@ var _ = Describe("odo storage command tests", func() {
...
-			helper.Cmd("odo", "push", "--context", commonVar.Context).ShouldPass()
-			StorageList = helper.Cmd("odo", "storage", "list", "--context", commonVar.Context).ShouldPass().Out()
-			Expect(StorageList).To(ContainSubstring("Pushed"))
-
-			// Delete storage, list storage should have state "Locally Deleted"
-			helper.Cmd("odo", "storage", "delete", "pv1", "-f", "--context", commonVar.Context).ShouldPass()
-			StorageList = helper.Cmd("odo", "storage", "list", "--context", commonVar.Context).ShouldPass().Out()
-			Expect(StorageList).To(ContainSubstring("Locally Deleted"))
-
+		When("creating storage", func() {
```
Comment (dharmit) [inline tests/integration/cmd_storage_test.go:85]: Why can't this be a part of the `When("creating storage")` block defined earlier? At least the `BeforeEach` in both cases is doing the same thing except using a different storage name and different `path`.

**Review by dharmit — CHANGES_REQUESTED** [review]

File: tests/integration/cmd_storage_test.go (hunk 5 above)
Comment (anandrkskd) [reply tests/integration/cmd_storage_test.go:85]: The component that we are creating here is different then what we are using in the above(L35) , I guess we can do that if we use the same component.

**Review by dharmit — COMMENTED** [review]
/approve

**Pull request merged**